### PR TITLE
Task/173517 add new fields to entity and create migration

### PR DESCRIPTION
**Pull request by pablocodeglia**
Description:
- Adds new fields to `innovation_reassessment_request` table, and the migration script to update it.
- Retrieves new fields on `InnovationAssessmentsService.getInnovationAssessmentInfo()`
- Updates `InnovationAssessmentsService.createInnovationReassessment()` to calculate `majorVersion` field, and removes cloning previous information to the new one.

Related tickets:

Closes [AB#173517](https://dev.azure.com/NHSiDev/InnovatorService/_workitems/edit/173517) & [AB#173748](https://dev.azure.com/NHSiDev/InnovatorService/_workitems/edit/173748)

US: [AB#168231](https://dev.azure.com/NHSiDev/InnovatorService/_workitems/edit/168231) & [AB#168243](https://dev.azure.com/NHSiDev/InnovatorService/_workitems/edit/168243)

### CODE REVIEW

File: apps/innovations/_types/innovation.types.ts [hunk 1]
```diff
@@ -21,12 +21,12 @@ export interface InnovationSectionModel {
...
 }
 
-export type ReassessmentType =
-  | {
-      updatedInnovationRecord: CurrentCatalogTypes.catalogYesNo;
-      description: string;
-    }
-  | { description: string };
+export type ReassessmentType = {
+  reassessmentReason: string[];
```
Comment (diogo-costa-bjss) [inline apps/innovations/_types/innovation.types.ts:25]: Shouldn't this have a type? With the valid values for this question?

File: apps/innovations/v1-innovation-reassessment-request-create/validation.schemas.ts [hunk 2]
```diff
@@ -11,19 +10,15 @@ export const ParamsSchema = Joi.object<ParamsType>({
...
-  {
-    is: ServiceRoleEnum.INNOVATOR,
-    then: Joi.object({
-      updatedInnovationRecord: Joi.string()
-        .valid(...Object.values(YesOrNoCatalogueType))
-        .required()
-    }).required()
-  }
-]);
+  reassessmentReason: Joi.array().items(Joi.string()).min(1).required(),
```
Comment (diogo-costa-bjss) [inline apps/innovations/v1-innovation-reassessment-request-create/validation.schemas.ts:20]: If you add the type of valid values, in here you should validate that only valid items are sent.

File: libs/shared/entities/innovation/innovation-reassessment-request.entity.ts [hunk 3]
```diff
@@ -17,6 +17,15 @@ export class InnovationReassessmentRequestEntity extends BaseEntity {
...
 
+  @Column({ name: 'reassessment_reason', type: 'simple-array' })
+  reassessmentReason: string[];
+
+  @Column({ name: 'other_reassessment_reason', type: 'nvarchar' })
+  otherReassessmentReason: string;
+
+  @Column({ name: 'what_support_do_you_need', type: 'nvarchar' })
+  whatSupportDoYouNeed: string;
+
```
Comment (diogo-costa-bjss) [inline libs/shared/entities/innovation/innovation-reassessment-request.entity.ts:28]: This values can be null right?
If yes, we need to add `nullable: true` to the decorator and on the type `string | null`.

**Review by diogo-costa-bjss — CHANGES_REQUESTED** [review]

File: libs/shared/entities/innovation/innovation-reassessment-request.entity.ts (hunk 3 above)
Comment (pablocodeglia) [reply libs/shared/entities/innovation/innovation-reassessment-request.entity.ts:28]: Right! Will fix

File: apps/innovations/_types/innovation.types.ts (hunk 1 above)
Comment (pablocodeglia) [reply apps/innovations/_types/innovation.types.ts:25]: Makes sense, will do.

File: apps/innovations/v1-innovation-reassessment-request-create/validation.schemas.ts (hunk 2 above)
Comment (pablocodeglia) [reply apps/innovations/v1-innovation-reassessment-request-create/validation.schemas.ts:20]: Will do.

**Review by diogo-costa-bjss — APPROVED** [review]

**Pull request merged**